src/seedu/addressbook/data/person/Person.java

```diff
@@ -134,8 +132,27 @@ public void removeAccount() {
         account = null;
     }
 
-    public List<Assessment> getAssessments() {
-        return new ArrayList<>(assessments);
+    public Set<Assessment> getAssessments() {
+        return new HashSet<>(assessments);
+    }
```

Comment:
Hmmm there's some naming issue here. You claim to be giving the Set of Assessment, but you are actually giving a copy of the set.
EDIT: I realized that it is also the same for the other getList function. I think it's something we need to root out as a team